packages/app-desktop/gui/NoteEditor/utils/contextMenuUtils.ts

```diff
@@ -0,0 +1,92 @@
+import Resource from '@joplin/lib/models/Resource';
+
+export enum ContextMenuItemType {
+	None = '',
+	Image = 'image',
+	Resource = 'resource',
+	Text = 'text',
+	Link = 'link',
+}
+
+export interface ContextMenuOptions {
+	itemType: ContextMenuItemType;
+	resourceId: string;
+	mime: string;
+	filename: string;
+	linkToCopy: string;
+	textToCopy: string;
+	htmlToCopy: string;
+	insertContent: Function;
+	isReadOnly?: boolean;
+}
+
+export interface ContextMenuItem {
+	label: string;
+	onAction: Function;
+	isActive: Function;
+}
+
+export interface ContextMenuItems {
+	[key: string]: ContextMenuItem;
+}
+
+export async function resourceInfo(options: ContextMenuOptions): Promise<any> {
+	const resource = options.resourceId ? await Resource.load(options.resourceId) : null;
+	const filePath = resource ? Resource.fullPath(resource) : null;
+	const filename = resource ? (resource.filename ? resource.filename : resource.title) : options.filename ? options.filename : '';
+	return { resource, filePath, filename };
+}
```

Comment:
@asrient, unfortunately you broke the desktop app context menu with this change. There are three problems:

1. The `resourceInfo()` returns "any", which means type checking is disabled for this function. Before you moved it here, the function also returned "any" but when you refactor code it's a good idea to try to improve it, especially since type-checking can save you a lot of troubles as it would have done in this case.

2. There's no tests for this function which again prevents you to know it's broken. Not entirely your fault either, since there was no tests before, but again when moving code around that's when you should analyse the code and test units are good for that.

3. And the bug is that the code in several places expects this function to return `resourcePath`, which you've removed from the function. So some context menu items are broken such as "Reveal in folder", "Save as", etc.

Any chance you could look into this and propose a fix?

More info in this thread: https://discourse.joplinapp.org/t/pre-release-v2-8-is-now-available-updated-14-april/25158/16?u=laurent

Reply:
Hi, really sorry for this, I will try to look into it and fix it asap.

Reply:
That would be great if you do. Thanks for your prompt response.

Reply:
Hi, I have posted a fix now https://github.com/laurent22/joplin/pull/6422. Sorry for the inconveniences caused.